### PR TITLE
Set default alert duration, and alternate between qr code and order details in Receipt page

### PR DESCRIPTION
**Pull request by redhoteggplant**
* Alerts had to be closed manually previously; now, "new" alerts disappear after a default duration
  * "New" alerts refer to when the `alertMessage` or `alertSeverity` state is updated, and the snackbar is open
* Receipt page shows QR code by default, and tapping it will show the order details instead (and vice versa)
  * This allows QR code to take a bigger portion of the screen, as per discussion with @devYaoYH 
ScanAndGo GPay (1)

### CODE REVIEW

File: client/src/components/AlertSnackbar/AlertSnackbar.tsx [hunk 1]
```diff
@@ -12,14 +13,39 @@ const AlertSnackbar = () => {
   const { open, setOpen, alertMessage, alertSeverity } = useContext(
     AlertContext
   );
+  const [duration, setDuration] = React.useState<number>(0);
   const anchorOrigin: SnackbarOrigin = {
     vertical: "top",
     horizontal: "center",
   };
 
+  const handleClose = (event?: React.SyntheticEvent, reason?: string) => {
```
Comment (cszqwe) [inline client/src/components/AlertSnackbar/AlertSnackbar.tsx:22]: Add some comments around the following lines. 
"open" is not a good parameter name as it is too generic, considering changing it to something more specific

File: client/src/components/AlertSnackbar/AlertSnackbar.tsx [hunk 2]
```diff
@@ -12,14 +13,39 @@ const AlertSnackbar = () => {
...
+    if (reason === "clickaway") {
```
Comment (cszqwe) [inline client/src/components/AlertSnackbar/AlertSnackbar.tsx:23]: Could you please add a comment to explain why there is a special check for "clickaway"?

File: client/src/components/AlertSnackbar/AlertSnackbar.tsx [hunk 3]
```diff
@@ -12,14 +13,39 @@ const AlertSnackbar = () => {
...
+  };
+
+  React.useEffect(() => {
+    if (!open) {
+      setDuration(0);
+    }
+  }, [open]);
+
+  React.useEffect(() => {
+    if (open && alertMessage) {
```
Comment (cszqwe) [inline client/src/components/AlertSnackbar/AlertSnackbar.tsx:36]: What would happen if 
open is true, while there is no alertMessage? Is this expected?

Comment (redhoteggplant) [reply client/src/components/AlertSnackbar/AlertSnackbar.tsx:36]: The alert would just show the severity icon with no message, which I'm presuming we would not ever intend, so a "new" alert is only when the alertMessage is non-empty

**Pull request merged**